src/sage/schemes/riemann_surfaces/riemann_surface.py

```diff
@@ -4012,107 +4011,110 @@ def vectomat(v):
+    EXAMPLES::
+        
+        sage: from sage.schemes.riemann_surfaces.riemann_surface import skew_form
+        sage: M = Matrix(ZZ, [
+        ....:             [ 0,  2,  0,  0,  0,  0, -2],
+        ....:             [-2,  0, -2,  0,  0,  0,  0],
+        ....:             [ 0,  2,  0,  6,  0,  0,  4],
+        ....:             [ 0,  0, -6,  0,  0,  0,  0],
+        ....:             [ 0,  0,  0,  0,  0,  0,  0],
+        ....:             [ 0,  0,  0,  0,  0,  0,  0],
+        ....:             [ 2,  0, -4,  0,  0,  0,  0],
+        ....:         ])
+        sage: M_prime, S = skew_form(M, transformation=True)
+        sage: M_prime
+        [ 0  2  0  0  0  0  0]
+        [-2  0  0  0  0  0  0]
+        [ 0  0  0  6  0  0  0]
+        [ 0  0 -6  0  0  0  0]
+        [ 0  0  0  0  0  0  0]
+        [ 0  0  0  0  0  0  0]
+        [ 0  0  0  0  0  0  0]
+        sage: bool(S.transpose() * M * S == M_prime)
+        True
+
+    """
+    # Must convert to cpython before moving to matrix_integer_dense
 
     nrows, ncols = M.nrows(), M.ncols()
-    if (nrows % 2) != 0 or (ncols % 2) != 0:
-        raise ValueError("M must be 2m x 2g with even dimensions.")
-    m = nrows // 2
-    g = ncols // 2
-
-    J = J_matrix(g)
-    K = M * J * M.transpose()
-    if (K + K.transpose()) != 0:
-        raise ValueError("K = M J M^T must be skew-symmetric over ZZ.")
-    if K.det() == 0:
-        raise ValueError("K must be non-degenerate (det(K) != 0).")
-
-    # block split
-    A = K[:m, :m]
-    B = K[:m, m : 2 * m]
-    D = K[m : 2 * m, m : 2 * m]
-
-    # fast path
-    if A.is_zero() and D.is_zero():
-        U, _, V = B.smith_form()
-
-        # S = diag(U^T, V) yields S^T K S = (0, delta; -delta, 0)
-        S = Matrix.block(
-            [[U.transpose(), Matrix.zero(ZZ, m, m)], [Matrix.zero(ZZ, m, m), V]]
-        )
-        M_clean = S.transpose() * M
-        return M_clean
+    if nrows != ncols:
+        raise ValueError("M must be a square matrix.")
+    pair_count = nrows // 2
 
-    # fix M so that K = (0, delta; -delta, 0)
-    n = K.nrows()
-    S = Matrix.identity(ZZ, n)
-    K_curr = Matrix(K)
+    total_size = M.nrows()
+    S = Matrix.identity(ZZ, total_size)
+    M_curr = Matrix(M)
     t = 0
 
-    while t < n:
+    while t < total_size:
         # Step 1: Choose pivot
         pivot = None
-        for i in range(t, n):
-            for j in range(i + 1, n):
-                if K_curr[i, j] != 0:
+        for i in range(t, total_size):
+            for j in range(i + 1, total_size):
+                if M_curr[i, j] != 0:
                     pivot = (i, j)
                     break
             if pivot:
                 break
-        if pivot is None:
+        else:
             break
 
         i, j = pivot
 
         # move pivot to (t, t+1)
         if i != t:
-            P = Matrix.identity(ZZ, n)
+            P = Matrix.identity(ZZ, total_size)
             P.swap_rows(i, t)
             P.swap_columns(i, t)
```

Comment:
It's not an essential change, but you could construct the same P here using 
```
P = elementary_matrix(ZZ, total_size, row1=i, row2=t)
```
which is slightly more streamlined and for larger matrix will be faster.